### PR TITLE
Audit: Check all names for a given email address. Retain errors.

### LINKED ISSUE #32: Match all email addresses
We have multiple signatures with the same email address, but different names. Usually someone signs, gets told its wrong and signs again with the name matching the git author. The check then says its still wrong, since it only looks at the name of the first matching email address - we consider duplicate signatures as invalid: https://github.com/jquery/jquery-license/blob/55ea56c1787a6f863b8bda2296fa171a3abb9fc7/lib/signatures.js#L87-L93

I think the above is the cause of the CLA failure here: https://github.com/jquery/globalize/pull/456

The simplest solution I can think of is to go through signatures in reverse order, that way newer signatures override older (invalid) signatures.

### PR DESCRIPTION
**Pull request by jzaefferer**
This replaces the hashed method with the hashedDetails method, which now
removes duplicate names.

Fixes #32
Closes #38
Ref #17

### CODE REVIEW

File: lib/signatures.js [hunk 1]
```diff
@@ -50,9 +50,16 @@ function getSignatures( key ) {
...
+	return Promise.all( [
+		module.exports.raw( keys.cla ),
+		module.exports.raw( keys.caa )
+	] )
+		.then( function( signatures ) {
+			return signatures[ 0 ].concat( signatures[ 1 ] );
+		} )
+		.then( function( signatures ) {
+			return validate( signatures );
+		} );
```
Comment (scottgonzalez) [inline lib/signatures.js]: This method no longer just returns errors as the name implies. This also has a lot of overlap with `getSignatures()`/`all()`; should probably consolidate the logic.

Comment (jzaefferer) [inline lib/signatures.js]: That's correct, I'm working on a replacement. Wanted to start by increasing test coverage on this method. I'm now rewriting the method return a different data structure. Once all uses are switched over, I'll remove the current `hashed` method.

I guess at this point the added tests are more interesting than the implementation changes...

Comment (scottgonzalez) [inline lib/signatures.js]: Ok, well the test seems fine.

**Comment by jzaefferer** [comment]
Updated with a new `hashedDetails` method. From the last commit:

> This keeps using the email address at key, but then adds a list of names and a list of errors for each. Good entries have one name and no errors. Okay entries have mutliple names and no errors. Bad entries have one or more names and one or more errors.

Unless this has some terrible flaw, next I'm going to update other parts of the program to make use of this new method.

**Comment by jzaefferer** [comment]
Updated once more, now its doing what its actually supposed to do, not just some intermediate step. Goes together with https://github.com/jquery/jquery-wp-content/pull/371

@scottgonzalez @arschmitz Could I have another review?

**Comment by arschmitz** [comment]
LGTM

**Comment by jzaefferer** [comment]
Updated to fix the case where a signature exists, but has errors. There was one PR that was indeed still invalid. I've also squashed everything into a single commit.

**Comment by jzaefferer** [comment]
Merged and tagged as 0.5.0. If something goes wrong while I'm on vacation the next 7 days, @arschmitz has access for emergency ops.

**Pull request merged**